tensorboard/plugins/graph/tf_graph_common/util.ts

```diff
@@ -106,14 +113,25 @@ module tf.graph.util {
     // UI to update.
     try {
       let result = tf.graph.util.time(msg, task);
-      // Update the progress value.
-      tracker.updateProgress(incProgressValue);
+
+      if (result instanceof Promise) {
```

Comment:
This code doesn’t make any sense to me. `runTask` is used for
synchronous computations. It computes the result synchronously, it
updates the progress synchronously, and it returns the result. Changing
this function such that if the result _happens_ to be a promise then the
time-updating behavior is deferred seems wrong to me. It’s like saying,
“we’ll compute this value for you, and if it happens to be a string then
we’ll also print it to the console.”

We have a separate `runAsyncPromiseTask`, which already supports
updating the progress and logging the timing data asynchronously. If
callers are using `runTask` where they should be using
`runAsyncPromiseTask`, then they are wrong and should be fixed. We
shouldn’t try to shoehorn “sometimes-synchronous sometimes-async”
behavior into an existing function that has a clear contract.

Reply:
I don't understand the need for two different methods, really, but that really is a separate task we can visit later. Done.